src/main/java/io/nats/client/ErrorListener.java

```diff
@@ -129,4 +130,46 @@ enum FlowControlSource { FLOW_CONTROL, HEARTBEAT }
      * @param source enum indicating flow control handling in response to which type of message
      */
     default void flowControlProcessed(Connection conn, JetStreamSubscription sub, String subject, FlowControlSource source) {}
+
+    /**
+     * Called by the connection when a low level socket write timeout occurs.
+     *
+     * @param conn The connection that had the issue
+     */
+    default void socketWriteTimeout(Connection conn) {}

```

Comment:
This is an ErrorListener event. It has a default implementation to be backward compatible.